### PR TITLE
Refactor tf BatchPolopt

### PR DESCRIPTION
**Pull request by AiRuiChen**
Complete fix of [#1391 ](https://github.com/rlworkgroup/garage/issues/1391).

TF version of BatchPolopt is now removed and inheritance relationship now looks cleaner.

### CODE REVIEW

File: src/garage/tf/_functions.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        max_path_length (int): Maximum length of a single rollout.
+        baseline_predictions(numpy.ndarray): : Predicted value of GAE
+            (Generalized Advantage Estimation) Baseline.
+        discount (float): Environment reward discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+
+    Returns:
+        dict: Processed sample data, with key
+            * observations: (numpy.ndarray)
```
Comment (ryanjulian) [inline src/garage/tf/_functions.py:23]: does this return numpy arrays or tf.tensors? If it returns numpy arrays, can't you just use `garage.np.paths_to_tensors`?

File: src/garage/tf/algos/npo.py [hunk 2]
```diff
@@ -156,13 +170,113 @@ def init_opt(self):
...
+            paths (list[dict]): A list of collected paths.
+
+        Returns:
+            numpy.float64: Average return.
+
+        """
+        # -- Stage: Calculate baseline
+        if self._flatten_input:
+            paths = [
+                dict(
```
Comment (ryanjulian) [inline src/garage/tf/algos/npo.py:208]: couldn't this be way shorter with a single helper variable?

File: src/garage/tf/algos/npo.py [hunk 3]
```diff
@@ -156,13 +170,113 @@ def init_opt(self):
...
+
+        Args:
+            paths (list[dict]): A list of collected paths.
+
+        """
+        logger.log('Logging diagnostics...')
+        self.policy.log_diagnostics(paths)
+        self._baseline.log_diagnostics(paths)
+
+    # pylint: disable=unused-argument
```
Comment (AiRuiChen) [inline src/garage/tf/algos/npo.py:272]: Parameter `itr` is not used in the original `BatchPolopt` implementation, yet files in the tf/ directory still follows this common signature.  See this [workaround attempt](https://github.com/rlworkgroup/garage/blob/master/src/garage/np/algos/off_policy_rl_algorithm.py#L138). 

We can create a separate PR to refactor this.

Comment (ryanjulian) [inline src/garage/tf/algos/npo.py:272]: is there a reason to not just remove itr?

**Pull request closed without merging**